code/datums/abilities/flock/flockmind.dm

```diff
@@ -204,27 +204,35 @@
 
 /datum/targetable/flockmindAbility/healDrone
 	name = "Concentrated Repair Burst"
-	desc = "Fully heal a drone through acceleration of its repair processes."
+	desc = "Heal all flock units in an area."
 	icon_state = "heal_drone"
-	cooldown = 20 SECONDS
+	cooldown = 30 SECONDS
+	var/max_targets = 4 //maximum number of drones healed
 
-/datum/targetable/flockmindAbility/healDrone/cast(mob/living/critter/flock/drone/target)
+/datum/targetable/flockmindAbility/healDrone/cast(atom/target)
 	if(..())
 		return TRUE
-	if(!istype(target))
-		return TRUE
-	if (target.get_health_percentage() >= 1)
-		boutput(holder.get_controlling_mob(), "<span class='notice'>[target.real_name] has no damage!</span>")
-		return TRUE
-	if (isdead(target))
-		boutput(holder.get_controlling_mob(), "<span class='notice'>[target.real_name] is dead!</span>")
-		return TRUE
+	var/healed = 0
+	for (var/mob/living/critter/flock/flockcritter in range(3, target))
+		var/health_ratio = flockcritter.get_health_percentage()
+		if (isdead(flockcritter) || health_ratio >= 1)
+			continue
+		flockcritter.HealDamage("All", 30, 30) //half of a flockdrone's health
```

Comment:
```suggestion
		flockcritter.HealDamage("All", flockcritter.health_brute / 2, flockcritter.health_burn / 2) //half of a flockdrone's health
```
Or similar, think this is full health currently?

Reply:
nope, for reasons I am not quite sure of, 30 brute and burn heals half the drone's HP

Reply:
Okay. Should probably still use the health_brute and health_burn variables since those are the actual health values for the flockcritters

Reply:
That would make flockbits healed for substantially less than drones, not sure why we'd want that

Reply:
Yeah and then there could be a separate check for flockbits

Reply:
but why?

Reply:
Err, I'm just saying that health_brute and health_burn can be used instead of hard coded values, for flockbits it'd be health_brute * 2 and health_burn * 2, and for flockdrones it'd be health_brute and health_burn

Reply:
Not sure if any more thought on this if it matters